src/triage/tracking.py

```diff
@@ -0,0 +1,282 @@
+import datetime
+import platform
+import getpass
+import os
+import requests
+import subprocess
+import logging
+from functools import wraps
+from triage.util.db import scoped_session
+from contextlib import contextmanager
+from triage import __version__
+
+
+from triage.component.results_schema import ExperimentRun, ExperimentRunStatus
+
+
+def infer_git_hash():
+    """Attempt to infer the git hash of the repository in the current working directory
+
+    Returns: Either the 'git rev-parse HEAD' output or None
+    """
+    try:
+        git_hash = subprocess.check_output(['git', 'rev-parse', 'HEAD']).strip().decode('utf-8')
+    except Exception as e:
+        logging.info("Unable to infer git hash, error: %s", str(e))
+        git_hash = None
+    return git_hash
+
+
+def infer_triage_version():
+    return __version__
+
+
+def infer_installed_libraries():
+    """Attempt to infer the installed libraries by running pip freeze and formatting as a list
+
+    Returns: Either a list, or None
+    """
+    try:
+        installed_libraries = subprocess.check_output(['pip', 'freeze']).split(b'\n')
+    except FileNotFoundError:
+        logging.info("Unable to pip freeze, cannot list installed libraries")
+        installed_libraries = []
+    return installed_libraries
+
+
+def infer_ec2_instance_type():
+    """Attempt to infer the instance type of the ec2 instance by querying Amazon's endpoint
+
+    Returns: Either the ec2 instance type as returned by Amazon's endpoint, or None
+    """
+    try:
+        ec2_instance_type = requests.get(
+            'http://169.254.169.254/latest/meta-data/instance-type',
+            timeout=1
```

Comment:
FYI: In Appy, the timeout for meta-data requests has always been `0.01`, without issue.